### PR TITLE
Add IconLayer demo to the gallery

### PR DESCRIPTION
**Pull request by kaushikb11**
For #2968 

- Add IconLayer demo to the gallery. Used auto packing to get Icons. In the example, avatars of Github contributors to the `deckgl` repo were used as Icons. 

- The `edgeLength` is scaled to the number of contributions made by the contributor.

Peek 2019-04-19 02-58

### CODE REVIEW

**Comment by xintongxia** [comment]
Great work! @kaushikb11 

A few suggestions,
1. Node `deck.gl` could use [website](http://deck.gl/#/) background.
2. Could try d3 `force-layout` to separate the nodes from each other, right now some of them are overlapped. (this could be a follow up PR)
3. I think correlate the icon size with the contributions would be more visible to the users. 
4. Could change the background to white and use a more muted color for edges.

**Comment by kaushikb11** [comment]
**Additional changes**

I have used the logic of quartiles and 90th percentile to make the correlation of icon size to the no of contributions more visible. As there were outliers, just scaling the contributions wasn't enough. I think even after using force layout when the links are concentrated, there was a bit of overlapping of icons because of Icon sizes. So, I made the links evenly distributed throughout the `main node`. There's almost no overlapping now.

Peek 2019-04-23 00-19

File: examples/gallery/src/icon-layer.html [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      #tooltip {
+        font-family: Helvetica, Arial, sans-serif;
+        font-size: 11px;
+        position: absolute;
+        padding: 4px;
+        margin: 8px;
+        background: rgba(0, 0, 0, 0.8);
+        color: #fff;
+        max-width: 300px;
+        font-size: 10px;
```
Comment (xintongxia) [inline examples/gallery/src/icon-layer.html]: duplicate

File: examples/gallery/src/icon-layer.html [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      const {firstQuartile, secondQuartile, thirdQuartile} = calculateQuartiles(listOfContributions);
+      const ninetiethPercentile = [...listOfContributions]
+        .sort((a, b) => a - b)[Math.floor(0.90 * listOfContributions.length)];
+
+      listOfContributions.forEach((number, index) => {
+        const linkDistance = 20 * ((number - Math.min(...listOfContributions))
+         / (Math.max(...listOfContributions) - Math.min(...listOfContributions))) + 30;
+
+        addTargetPositionToContributor(data[index], linkDistance);
+      })
```
Comment (xintongxia) [inline examples/gallery/src/icon-layer.html]: In your code, you calculate the min and max for every contribution, which is not good for performance.

You can use a temporary variable to store this `Math.min(...listOfContributions)` and `Math.max(...listOfContributions)`

**Comment by xintongxia** [comment]
This is great. We can have a follow up PR to make this layout more interactive.

File: examples/gallery/src/icon-layer.html [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+      const maxContributions = Math.max(...listOfContributions);
+      const minContributions = Math.min(...listOfContributions);
+
+      listOfContributions.forEach((number, index) => {
+        const linkDistance = 20 * ((number - minContributions) 
+        / (maxContributions - minContributions)) + 30;
+
+        addTargetPositionToContributor(data[index], linkDistance);
+      })
```
Comment (xintongxia) [inline examples/gallery/src/icon-layer.html]: Could extract the logic to get contributor nodes' positions to a separate function.

**Pull request merged**